pandas/core/interchange/column.py

```diff
@@ -278,20 +278,28 @@ def _get_data_buffer(
         """
         Return the buffer containing the data and the buffer's associated dtype.
         """
-        if self.dtype[0] in (
-            DtypeKind.INT,
-            DtypeKind.UINT,
-            DtypeKind.FLOAT,
-            DtypeKind.BOOL,
-            DtypeKind.DATETIME,
-        ):
+        if self.dtype[0] == DtypeKind.DATETIME:
             # self.dtype[2] is an ArrowCTypes.TIMESTAMP where the tz will make
             # it longer than 4 characters
-            if self.dtype[0] == DtypeKind.DATETIME and len(self.dtype[2]) > 4:
+            if len(self.dtype[2]) > 4:
```

Comment:
Not sure the history of the comment but as far as I am aware the extra metadata attached to a timestamp type in Arrow is a blank string for tz-naive, and anything else for tz-aware. `"utc"` is a valid timezone name which if I read this correctly would be considered naive through this interface?

Reply:
The format string is of the form "tsn:<time zone>" https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings

So, if there's no time zone, it'll just be `"tsn:"`, whereas for UTC it would be `"tsn:UTC"`:
```python
> /home/marcogorelli/pandas-dev/pandas/core/interchange/column.py(285)_get_data_buffer()
-> if len(self.dtype[2]) > 4:
(Pdb) p self.dtype[2]
'tsn:'
> /home/marcogorelli/pandas-dev/pandas/core/interchange/column.py(285)_get_data_buffer()
-> if len(self.dtype[2]) > 4:
(Pdb) p self.dtype[2]
'tsn:UTC'
```